File: 01-Arrays/20250718-beginner/ts/solution.ts
/**
 * Find the maximum element in an array
 * @param arr Array of numbers
 * @returns The maximum number in the array
 * @throws Error if array is empty
 */
export function findMaxElement(arr: number[]): number {
    // TODO: Implement your solution here
    if (arr.length < 1) {
        throw new Error("Method not implemented");
    }
    let max: number | null = null;
    for (let index = 0; index < arr.length; index++) {
        if (index === 0) {
            max = arr[index]
        }
        const element = arr[index];
        if (max !== null && max < element) {
            max = element
        }
    }
    if (max === null) {
        throw new Error("Max cannot be null")
    }
    return max
}

File: 01-Arrays/20250718-beginner/go/solution.go
package main

import "errors"

// FindMaxElement finds the maximum element in an array of integers
// Returns an error if the array is empty
func FindMaxElement(arr []int) (int, error) {
	// TODO: Implement your solution here
	if len(arr) < 1 {
		return 0, errors.New("method not implemented")
	}
	var max int
	for index, item := range arr {
		if index == 0 {
			max = item
		}
		if item > max {
			max = item
		}
	}
	return max, nil
}
